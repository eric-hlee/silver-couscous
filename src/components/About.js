import React, { useState, useEffect } from "react";
import { FormattedMessage } from "react-intl";
import Tabs from "react-bootstrap/Tabs";
import Tab from "react-bootstrap/Tab";
import WorkExperience from "./WorkExperience";
import Education from "./Education";
import KeySkills from "./KeySkills";

export default function About() {
	const [workDescriptions, setWorkDescriptions] = useState({
		conversocial: null,
		verv: null,
		ibm: null,
		queensu: null,
	});

	const getWorkDescriptions = async () => {
		let response = await fetch("https://morning-garden-95959.herokuapp.com/about");
		let data = await response.json();
		return data;
	};

	useEffect(() => {
		getWorkDescriptions()
			.then((res) => res[0])
			.then((res) =>
				setWorkDescriptions({
					conversocial: res.conversocial,
					verv: res.verv,
					ibm: res.ibm,
					queensu: res.queensu,
				})
			)
			.catch((err) => {
				console.log(err);
			});
	}, []);

	return (
		<main>
			<Tabs defaultActiveKey="work-experience" id="about_tabs">
				<Tab
					eventKey="work-experience"
					title={
						<FormattedMessage
							id="about.work_experience"
							defaultMessage="ðŸ“ƒ Work Experience"
						/>
					}
				>
					<WorkExperience workDescriptions={workDescriptions} />
				</Tab>
				<Tab
					eventKey="education"
					title={
						<FormattedMessage
							id="about.education"
							defaultMessage="ðŸŽ“ Education"
						/>
					}
				>
					<Education />
				</Tab>
				<Tab
					eventKey="key-skills"
					title={
						<FormattedMessage
							id="about.key_skills"
							defaultMessage="ðŸ§° Key Skills"
						/>
					}
				>
					<KeySkills />
				</Tab>
			</Tabs>
		</main>
	);
}
