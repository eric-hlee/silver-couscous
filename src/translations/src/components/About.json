[{"id":"about.work_experience","defaultMessage":"📃 Work Experience","file":"src/components/About.js","start":{"line":59,"column":12},"end":{"line":62,"column":14}},{"id":"about.education","defaultMessage":"🎓 Education","file":"src/components/About.js","start":{"line":73,"column":12},"end":{"line":76,"column":14}},{"id":"about.key_skills","defaultMessage":"🧰 Key Skills","file":"src/components/About.js","start":{"line":84,"column":12},"end":{"line":87,"column":14}}]
