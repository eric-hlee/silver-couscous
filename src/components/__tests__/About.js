import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { IntlProvider } from "react-intl";
import { shallow } from "enzyme";
import About from "../About";

describe("<About />", () => {
	let container = null;
	beforeEach(() => {
		// setup a DOM element as a render target
		container = document.createElement("div");
		document.body.appendChild(container);
	});

	afterEach(() => {
		// cleanup on exiting
		unmountComponentAtNode(container);
		container.remove();
		container = null;
	});

	it("renders correctly", () => {
		const wrapper = shallow(<About />);
		expect(wrapper).toMatchSnapshot();
	});

	it("should return data with a successful request", async () => {
		const mockResponse = [
			{
				conversocial: ["work", "work", "work"],
				verv: ["work", "work", "work"],
				ibm: ["work", "work", "work"],
				queensu: ["work", "work", "work"],
			},
		];
		jest.spyOn(global, "fetch").mockImplementation(() => {
			return Promise.resolve({
				json: () => Promise.resolve(mockResponse),
			});
		});

		// Use the asynchronous version of act to apply resolved promises
		await act(async () => {
			render(
				<IntlProvider locale="en">
					<About />
				</IntlProvider>,
				container
			);
		});

		expect(container).toMatchSnapshot();
		expect(container.querySelector(".conversocial").textContent).toContain(
			mockResponse[0].conversocial.join("")
		);
		expect(container.querySelector(".verv").textContent).toContain(
			mockResponse[0].verv.join("")
		);
		expect(container.querySelector(".ibm").textContent).toContain(
			mockResponse[0].ibm.join("")
		);
		expect(container.querySelector(".queensu").textContent).toContain(
			mockResponse[0].queensu.join("")
		);

		// remove the mock to ensure tests are completely isolated
		global.fetch.mockRestore();
	});
});
