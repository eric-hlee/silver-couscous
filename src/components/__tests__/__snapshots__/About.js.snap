// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders correctly 1`] = `
<section>
  <Tabs
    defaultActiveKey="work-experience"
    id="about_tabs"
    mountOnEnter={false}
    unmountOnExit={false}
    variant="tabs"
  >
    <Tab
      className="loading"
      eventKey="work-experience"
      title={
        <FormattedMessage
          defaultMessage="ðŸ“ƒ Work Experience"
          id="about.work_experience"
        />
      }
    >
      <Spinner
        animation="border"
        className="spinner"
        role="status"
        variant="secondary"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </Spinner>
    </Tab>
    <Tab
      eventKey="education"
      title={
        <FormattedMessage
          defaultMessage="ðŸŽ“ Education"
          id="about.education"
        />
      }
    >
      <Education />
    </Tab>
    <Tab
      eventKey="key-skills"
      title={
        <FormattedMessage
          defaultMessage="ðŸ§° Key Skills"
          id="about.key_skills"
        />
      }
    >
      <KeySkills />
    </Tab>
  </Tabs>
</section>
`;

exports[`<About /> should return data with a successful request 1`] = `
<div>
  <section>
    <nav
      class="nav nav-tabs"
      role="tablist"
    >
      <a
        aria-controls="about_tabs-tabpane-work-experience"
        aria-selected="true"
        class="nav-item nav-link active"
        data-rb-event-key="work-experience"
        href="#"
        id="about_tabs-tab-work-experience"
        role="tab"
      >
        ðŸ“ƒ Work Experience
      </a>
      <a
        aria-controls="about_tabs-tabpane-education"
        aria-selected="false"
        class="nav-item nav-link"
        data-rb-event-key="education"
        href="#"
        id="about_tabs-tab-education"
        role="tab"
        tabindex="-1"
      >
        ðŸŽ“ Education
      </a>
      <a
        aria-controls="about_tabs-tabpane-key-skills"
        aria-selected="false"
        class="nav-item nav-link"
        data-rb-event-key="key-skills"
        href="#"
        id="about_tabs-tab-key-skills"
        role="tab"
        tabindex="-1"
      >
        ðŸ§° Key Skills
      </a>
    </nav>
    <div
      class="tab-content"
    >
      <div
        aria-hidden="false"
        aria-labelledby="about_tabs-tab-work-experience"
        class="fade tab-pane active show"
        id="about_tabs-tabpane-work-experience"
        role="tabpanel"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li />
            <li />
            <li
              class="active"
            />
            <li />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item"
            >
              <div
                class="card work"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="media"
                  >
                    <img
                      alt="Conversocial logo"
                      class="mr-3"
                      height="64"
                      src="test-file-stub"
                      width="64"
                    />
                    <div
                      class="media-body"
                    >
                      <div
                        class="card-title h5"
                      >
                        <a
                          class="card-link"
                          href="http://www.conversocial.com/"
                          rel="noopener"
                          target="_blank"
                        >
                          Conversocial
                        </a>
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        Junior Software Engineer
                        <span
                          class="info-item"
                        >
                          Â·
                        </span>
                        May 2019 â€“ Apr 2020
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        London, United Kingdom
                      </div>
                      <div
                        class="conversocial list-group list-group-flush"
                      >
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <div
                class="card work"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="media"
                  >
                    <img
                      alt="Verv logo"
                      class="mr-3"
                      height="64"
                      src="test-file-stub"
                      width="64"
                    />
                    <div
                      class="media-body"
                    >
                      <div
                        class="card-title h5"
                      >
                        <a
                          class="card-link"
                          href="http://verv.energy/"
                          rel="noopener"
                          target="_blank"
                        >
                          Verv
                        </a>
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        Technical Support Engineer
                        <span
                          class="info-item"
                        >
                          Â·
                        </span>
                        Oct 2018 â€“ Apr 2019
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        London, United Kingdom
                      </div>
                      <div
                        class="verv list-group list-group-flush"
                      >
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="active carousel-item"
            >
              <div
                class="card work"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="media"
                  >
                    <img
                      alt="IBM logo"
                      class="mr-3"
                      height="64"
                      src="test-file-stub"
                      width="64"
                    />
                    <div
                      class="media-body"
                    >
                      <div
                        class="card-title h5"
                      >
                        <a
                          class="card-link"
                          href="http://www.ibm.com/"
                          rel="noopener"
                          target="_blank"
                        >
                          IBM
                        </a>
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        e-Commerce Java Development Support Intern
                        <span
                          class="info-item"
                        >
                          Â·
                        </span>
                        May 2016 â€“ Apr 2018
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        Toronto, Canada
                      </div>
                      <div
                        class="ibm list-group list-group-flush"
                      >
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <div
                class="card work"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="media"
                  >
                    <img
                      alt="Queen's University logo"
                      class="mr-3"
                      height="64"
                      src="test-file-stub"
                      width="64"
                    />
                    <div
                      class="media-body"
                    >
                      <div
                        class="card-title h5"
                      >
                        <a
                          class="card-link"
                          href="http://www.queensu.ca/"
                          rel="noopener"
                          target="_blank"
                        >
                          Queen's University
                        </a>
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        Teaching Assistant
                        <span
                          class="info-item"
                        >
                          Â·
                        </span>
                        Sep 2017 â€“ Apr 2018
                      </div>
                      <div
                        class="mb-2 text-muted card-subtitle h6"
                      >
                        Kingston, Ontario
                      </div>
                      <div
                        class="queensu list-group list-group-flush"
                      >
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                        <div
                          class="list-group-item"
                        >
                          work
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="about_tabs-tab-education"
        class="fade tab-pane"
        id="about_tabs-tabpane-education"
        role="tabpanel"
      >
        <div
          class="card"
        >
          <a
            class="card-link"
            href="http://www.queensu.ca/"
            rel="noopener"
            target="_blank"
          >
            <img
              class="card-img-top"
              src="test-file-stub"
            />
          </a>
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Queen's University
            </div>
            <p
              class="card-text"
            >
              <strong>
                Bachelor of Computing (Honours)
              </strong>
              <br />
              Specialization in Software Design with Professional Internship
              <br />
              Deanâ€™s Honour List 2017-18
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="about_tabs-tab-key-skills"
        class="fade tab-pane"
        id="about_tabs-tabpane-key-skills"
        role="tabpanel"
      >
        <div
          class="key-skills card"
        >
          <div
            class="card-body"
          >
            <div
              class="card-text"
            >
              <div
                class="card-title h5"
              >
                Programming languages:
              </div>
              <div
                class="list-group list-group-horizontal-sm"
              >
                <div
                  class="list-group-item"
                >
                  Python
                </div>
                <div
                  class="list-group-item"
                >
                  Java
                </div>
                <div
                  class="list-group-item"
                >
                  Bash
                </div>
              </div>
              <br />
              <div
                class="card-title h5"
              >
                Web/scripting languages:
              </div>
              <div
                class="list-group list-group-horizontal-sm"
              >
                <div
                  class="list-group-item"
                >
                  HTML
                </div>
                <div
                  class="list-group-item"
                >
                  CSS
                </div>
                <div
                  class="list-group-item"
                >
                  Sass
                </div>
                <div
                  class="list-group-item"
                >
                  JavaScript
                </div>
                <div
                  class="list-group-item"
                >
                  SQL
                </div>
              </div>
              <br />
              <div
                class="card-title h5"
              >
                Library/Frameworks:
              </div>
              <div
                class="list-group list-group-horizontal-sm"
              >
                <div
                  class="list-group-item"
                >
                  React
                </div>
                <div
                  class="list-group-item"
                >
                  Redux
                </div>
                <div
                  class="list-group-item"
                >
                  React-Router
                </div>
                <div
                  class="list-group-item"
                >
                  React Intl
                </div>
                <div
                  class="list-group-item"
                >
                  Node.js
                </div>
                <div
                  class="list-group-item"
                >
                  Draft.js
                </div>
                <div
                  class="list-group-item"
                >
                  Jest
                </div>
                <div
                  class="list-group-item"
                >
                  Enzyme
                </div>
              </div>
              <br />
              <div
                class="card-title h5"
              >
                Other technologies:
              </div>
              <div
                class="list-group list-group-horizontal-sm"
              >
                <div
                  class="list-group-item"
                >
                  Docker
                </div>
                <div
                  class="list-group-item"
                >
                  Git
                </div>
                <div
                  class="list-group-item"
                >
                  GitHub
                </div>
                <div
                  class="list-group-item"
                >
                  Webpack
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
